axelrod/strategies/mutual.py

```diff
@@ -0,0 +1,67 @@
+from axelrod import Actions, Player, init_args
+from axelrod.random_ import random_choice
+
+C, D = Actions.C, Actions.D
+
+
+class Desperate(Player):
+    """A player that only cooperates after mutual defection"""
+    name = "Desperate"
+    classifier = {
+        'memory_depth': 1,
+        'long_run_time': False,
+        'stochastic': True,
+        'makes_use_of': set(),
+        'inspects_source': False,
+        'manipulates_source': False,
+        'manipulates_state': False
+    }
+
+    def strategy(self, opponent):
+        if not opponent.history:
+            return random_choice()
+        if self.history[-1] == D and opponent.history[-1] == D:
+            return C
+        return D
+
+
+class Hopeless(Player):
+    """A player that only defects after mutual cooperation"""
+    name = "Hopeless"
+    classifier = {
+        'memory_depth': 1,
+        'long_run_time': False,
+        'stochastic': True,
+        'makes_use_of': set(),
+        'inspects_source': False,
+        'manipulates_source': False,
+        'manipulates_state': False
+    }
+
+    def strategy(self, opponent):
+        if not opponent.history:
+            return random_choice()
+        if self.history[-1] == C and opponent.history[-1] == C:
+            return D
+        return C
+
+
+class Willing(Player):
+    """A player that only defects after mutual defection"""
```

Comment:
How do you feel about adding the references here, so in each of the strategies adding:

```
Names:

- Willing: [Berg2015]_
```

and in bibliography.rst adding:

```
.. [Berg2015] Berg, P. Van Den, & Weissing, F. J. (2015). The importance of mechanisms for the evolution of cooperation. Proceedings of the Royal Society B-Biological Sciences, 282.
```

If you'd rather I'm happy to just do that in #225 but will be good to getting used to asking for these with new PRs :)